enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,270 @@
+---
+title: automated-backups-no-config
+authors:
+  - "@elbehery"
+reviewers:
+  - "@soltysh"
+  - "@dusk125"
+  - "@hasbro17"
+  - "@tjungblu"
+  - "@williamcaban"
+approvers:
+  - "@soltysh"
+  - "@dusk125"
+  - "@hasbro17"
+  - "@tjungblu"
+  - "@williamcaban"
+api-approvers:
+  - "@soltysh"
+creation-date: 2024-06-17
+last-updated: 2024-06-17
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-609
+see-also:
+  - "https://issues.redhat.com/browse/OCPSTRAT-1411"
+  - "https://issues.redhat.com/browse/OCPSTRAT-529"
+---
+
+
+# Automated Backups of etcd with No Config
+
+## Summary
+
+To enable automated etcd backups of an Openshift cluster from day one after installation, 
+without provided configuration from the user.
+
+## Motivation
+
+The current [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd) of an OpenShift
+cluster relies on user provided configurations. 
+
+To improve the user's experience, this work proposes taking etcd backup using default config from day one, without additional configuration from the user.
+
+### Goals
+
+* Backups should be taken without configuration after cluster installation from day 1.
+* Backups are saved to a default location on each master node's disk using `hostPath` volume.
+* Backups are taken according to a default schedule.
+* Backups are being maintained according to a default retention policy.
+
+### Non-Goals
+
+* Save cluster backups to remote cloud storage (e.g. S3 Bucket).
+  - This could be a future enhancement or extension to the API.
+  - Having backups in an independent storage from OCP cluster is crucial for Disaster Recovery scenarios.
+* Automate cluster restoration.
+* Provide automated backups for non-self hosted architectures like Hypershift.
+
+## Proposal
+
+To enable automated etcd backups of an Openshift cluster from Day 1, without user supplied configuration, this work proposes a default etcd backup option.
+
+The default backup option utilizes a sidecar container within each etcd pod. The container is responsible for maintaining backups within a `hostPath` volume on each master node's disk.
+
+This approach guarantees have backups across all master nodes which is vital in disaster recovery scenarios.
+
+### User Stories
+
+* As a cluster administrator, I would like OCP cluster backup to be taken without configuration from day one.
+* As a cluster administrator, I would like to schedule recurring OCP cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes).
+
+### API Extensions
+
+No [API](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions) changes are required.
```

Comment:
How would a user configure the destination of the backups without an API? If they wanted to move them to persistent storage? Looks like we are leveraging an existing API, so perhaps it's worth mentioning that this is re-using the existing API at this stage, rather than saying there is no API?

Reply:
>Looks like we are leveraging an existing API, so perhaps it's worth mentioning that this is re-using the existing API at this stage, rather than saying there is no API?

@Elbehery I think this is currently a source of confusion in this enhancement. Either we're using the existing `config.openshift.io/v1alpha1` `Backup` API or not? This seems to be saying it bothways.
>To distinguish default no-config backups from user configured counterparts, a config.openshift.io/v1alpha1 Backup CR's name must be default

Also relevant to my other comment is that this also brings into question whether or not this feature should have it's own feature gate `AutomatedBackupNoConfig ` when you're saying this is reliant on the API behind the existing `AutomatedEtcdBackup` feature gate.

https://github.com/openshift/enhancements/pull/1646/files#r1781963569

Reply:
> How would a user configure the destination of the backups without an API? 

I thought the goal of this epic was to have **no configuration**, why would that need an API?

Reply:
@JoelSpeed so we are using the same API introduced in `automated-backups` work. 

> re-using the existing API at this stage, rather than saying there is no API?

+1, I am changing this now. 

So what I meant that we did not introduce any API changes from `automated-backups`

Reply:
@hasbro17 

- Yes, we are using the same API `config.openshift.io/v1alpha1` `Backup`, I will clarify this now all over the text 👍🏽 
-  So as discussed before, we agreed to use the same feature gate, that's why we are utilizing the `PeriodicBackupController`. Also having two separate feature gates, imho;  will cause problems in the future as this is the same work.